src/agent/communicator/include/http_socket.hpp

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <ihttp_socket.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/asio/experimental/awaitable_operators.hpp>
+#include <boost/beast.hpp>
+#include <boost/system/error_code.hpp>
+
+#include <string>
+
+namespace http_client
+{
+    /// @brief Implementation of IHttpSocket for HTTP requests
+    class HttpSocket : public IHttpSocket
+    {
+    public:
+        /// @brief Constructor for HttpSocket
+        /// @param io_context The io context to use for the socket
+        HttpSocket(const boost::asio::any_io_executor& io_context)
+            : m_socket(io_context)
+        {
+        }
+
+        /// @brief Connects the socket to the given endpoints
+        /// @param io_context The io_context associated to the socket
+        /// @param endpoints The endpoints to connect to
+        /// @param ec The error code, if any occurred
+        /// @param timeOut The timeout for the connection
+        void Connect(boost::asio::io_context& ioContext,
+                     const boost::asio::ip::tcp::resolver::results_type& endpoints,
+                     boost::system::error_code& ec,
+                     const std::chrono::seconds timeOut = std::chrono::seconds(TIMEOUT_DEFAULT)) override;
```

Comment:
There's something that we should improve, this class takes in its constructor an io context, but we need to pass another one for this method and others below.